src/main/java/io/vertx/core/http/impl/HttpServerImpl.java

```diff
@@ -676,6 +676,7 @@ protected void handleMessage(Http1xServerConnection conn, ContextImpl context, C
             chctx.writeAndFlush(new PongWebSocketFrame(wsFrame.getBinaryData().copy()));
             break;
           case CLOSE:
+            conn.handleMessage(msg);
```

Comment:
rather do that after following statement, so a failure in the conn.handleMessage does not prevent the frame echo to be sent.